ArduPlane/tailsitter.cpp

```diff
@@ -218,6 +230,15 @@ void QuadPlane::tailsitter_speed_scaling(void)
         scaling = constrain_float(hover_throttle / throttle, 0, tailsitter.throttle_scale_max);
     }
 
+    // if no airspeed sensor reduce throws at large tilt angles (implies high airspeed)
+    if (!ahrs.airspeed_sensor_enabled()) {
+        const float angle_attenuation = 0.5f;
+        Matrix3f rot = ahrs_view->get_rotation_body_to_ned();
+        float tilt_angle = degrees(fabsf(acosf(rot.c.z)));
+        if (tilt_angle > tailsitter.transition_angle) {
+            scaling = constrain_float(angle_attenuation * (float)(tailsitter.transition_angle) / tilt_angle, 0.25f, 1.0f);
```

Comment:
should this not be *= other wise it overwrite the throttle scaling? I think you can get the pitch directly from the ahrs using ahrs_view->pitch, Why use acos not cos? I think ahrs_view->cos_pitch * tailsitter.transition_angle would be equivalent? Using the if statemnt you get a jump in scaling, it goes from 1 to 0.5 as tilt_angle becomes greater than transition_angle, having a  jump would upset he PID's. I will plot the scaling ratio vs tilt angle to get a better undersanding of whats going on.

Reply:
Overriding the throttle scaling is intentional

rot.c.z is the cosine of the angle between earth z and body z, so the arccos is the angle to vertical

And although I don't think step changes in PID gains cause any problems, I believe that transition_angle/tilt_angle starts near 1 and decreases as tilt_angle increases.

Reply:
![angle_scaling](https://user-images.githubusercontent.com/33176108/51695484-11f48280-1ffb-11e9-9962-e97a299735c9.png)

ah, that makes more sense, I guess you doing it that way to get the highest lean angle? A combination of roll and pitch rather than just pitch? I have plotted it, I hope correctly,  above assuming a scaling of one before it 'activates' does it look how you would expect?

Reply:
oh, that's right.. I had forgotten the attenuation_factor of 0.5
I'll try changing it so that the scaler goes from 1 down to attenuation_factor instead, but I'm not seeing any problems with this version in SITL

Reply:
I just pushed a change which makes the scaling linear from 1 at transition_angle down to 0.25 at q_angle_max
Works well for Stryker_quad with angle_max at 80 degrees

Reply:
![angle_scaling](https://user-images.githubusercontent.com/33176108/51704474-13c94080-2011-11e9-8bd1-f76bed298cf1.png)
Gives this now, not so far from just doing cos (in red). I still wonder if its a good idea, a example of where this would not work well would be taking off in a TVBS. You would be a 90deg of pitch and zero airspeed and need full range of movement to takeoff, this would limit you to 25%. Or a transition at a very low airspeed you may not be left with enough gain to get vertical with a none vectored tailsitter. I will try a few different models with realflight.

Reply:
I was only thinking of copter tailsitters here; perhaps there should be different rules for tiltrotors and  2 motor tailsitters.

What exactly are you plotting above? y=mx+b should be linear.

Reply:
Also, doesn't this scaling affect only the control surfaces? If so, a TVBS takeoff should be unaffected since there is no airflow at that point. That's the case at launch for the Stryker_quad also

Reply:
it is for the tilt motors also, no reason why you could't apply the new scaling to just the control surfaces tho, I will have a play about in real flight.